crates/agentgateway/proto/resource.proto

```diff
@@ -521,6 +521,14 @@ message PolicySpec {
     string expression = 1;
   }
 
+  // Cross-Site Request Forgery (CSRF) policy configuration.
+  message CSRF {
```

Comment:
Let's also add the conversion logic for csrf in https://github.com/agentgateway/agentgateway/blob/main/crates/agentgateway/src/types/agent_xds.rs. We'll want the additional origins to be a set of unique strings (dedup it as part of the translation)